### PR TITLE
Papermill api

### PR DESCRIPTION
**Pull request by edublancas**
closes https://github.com/ploomber/ploomber-engine/issues/24

added a bunch of missing features so we can be a drop-in replacement for papermill.

### CODE REVIEW

File: src/ploomber_engine/ipython.py [hunk 1]
```diff
@@ -146,6 +164,19 @@ class PloomberClient:
     display_output : bool, default=False
         If True, it prints the same output the notebook prints.
 
+    progress_bar : bool, default=True
+        Display a progress bar.
+
+    debug_later : bool or str, default=False
```
Comment (yafimvo) [inline src/ploomber_engine/ipython.py:170]: I think there can be confusion if `debug_later` can be a bool or a path

File: tests/test_util.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+@pytest.mark.parametrize(
+    "nb, params_idx, cells_total",
+    [
+        [_make_nb_obj(["1 + 1"]), 0, 2],
+        [_make_nb_obj([("code", "1 + 1", {"tags": ["parameters"]})]), 1, 2],
+        [_make_nb_obj(["21 + 21", ("code", "1 + 1", {"tags": ["parameters"]})]), 2, 3],
+        [_make_nb_obj(["# parameters\n1 + 2"]), 1, 2],
+        [_make_nb_obj(["1 + 1", "# parameters\n1 + 2"]), 2, 3],
+        [_make_nb_obj(["#PARAMETERS\n1 + 2"]), 1, 2],
+        [_make_nb_obj(["1 + 1", "#PARAMETERS\n1 + 2"]), 2, 3],
```
Comment (yafimvo) [inline tests/test_util.py:16]: Since users might use `# parameters`, I think we can add a test for lowercase as well.

**Review by yafimvo — CHANGES_REQUESTED** [review]

File: src/ploomber_engine/_util.py [hunk 3]
```diff
@@ -10,3 +18,118 @@ def recursive_update(target, update):
...
+    out = find_cell_with_tags(nb, [tag])
+
+    if out:
+        located = out[tag]
+        return located["cell"], located["index"]
+    else:
+        return None, None
+
+
+def find_cell_with_comment(nb):
```
[the remark this answers is not in the document]
Comment (edublancas) [reply src/ploomber_engine/_util.py:57]: I added the docstring and renamed it for `find_cell_with_parameters_comment`, I didn't add the `comment` argument this that would complicate the implementation and we don't have a need right now for finding other comments

File: src/ploomber_engine/ipython.py (hunk 1 above)
Comment (edublancas) [reply src/ploomber_engine/ipython.py:170]: i thought about it, but adding two parameters (one bool, and one path) seems too much. thoughts?

File: tests/test_util.py (hunk 2 above)
Comment (edublancas) [reply tests/test_util.py:16]: there is one lowercase test in line 14, do you think we need more?

Comment (yafimvo) [reply tests/test_util.py:16]: didn't notice it. definitely enough.

File: src/ploomber_engine/ipython.py (hunk 1 above)
Comment (yafimvo) [reply src/ploomber_engine/ipython.py:170]: I see your point.
I think if someone needs to store the output in a different path/file, he wouldn't mind writing an "extra" argument for specifying the path (maybe '--save' ?)

Comment (edublancas) [reply src/ploomber_engine/ipython.py:170]: I think let's give it a try. This API is not the primary one (only for advanced users). and it'll be easier to migrate them to a new one that has the extra parameter than the other way around.

**Pull request merged**